035-cluster-wide-volume-usage-quota-managemnet.md

```diff
@@ -0,0 +1,277 @@
+ */
+
+package io.strimzi.kafka.quotas;
+
+import java.util.concurrent.TimeUnit;
+
+public interface DataSourceTask extends Runnable {
+    @Override
+    void run();
+
+    long getPeriod();
+
+    TimeUnit getPeriodUnit();
+}
+```
+
+##### Message schema
+For simplicity and debug ability the message should be encoded as JSON, but could be converted to a more space efficient
+format later on if justified.
+
+###### Message
+
+| Field      | Type                                    | Description                                                                   |
+|------------|-----------------------------------------|-------------------------------------------------------------------------------|
+| SnapshotAt | `ISO 8601 Date Time`                    | Timestamp when the snapshot of volume usage was generated [1]                 |
+| Hard limit | [Limit Definition](#limit-definition)   | Defines when the source broker believes the producers should be paused [2]    |
+| Soft limit | `Limit Definition`                      | Defines when the source broker believes the producers should be throttled [2] |
+| Volumes    | Set of [VolumeDetails](#volume-details) | Details of the storage volumes attached to the broker                         |
+
+###### Limit Definition
+
+| Field | Type   | Description                                                   |
+|-------|--------|---------------------------------------------------------------|
+| type  | Enum   | One of `ConsumedSpace`, `MinFreeBytes` or `MinFreePercentage` |
+| level | Number | Defines the level at which the limit applies                  |
+
+###### Volume Details
+
+| Field      | Type   | Description                                                      |
+|------------|--------|------------------------------------------------------------------|
+| VolumeName | String | An identifier to uniquely distinguish the disk. e.g. `/dev/sda1` |
+| Capacity   | long   | The capacity of the volume in bytes                              |
+| Consumed   | long   | The number of bytes currently in use on the volume               |
+
+Notes:
+
+1. Explicitly mark when the volume usage was captured rather than depending on message publication timestamps.
+2. One could view the limits defined at the message level as being broker wide defaults, they could then optionally be
+   overridden at the VolumeDetails level with a volume specific limit.
+3. Separating freshness from topic eviction as freshness is a "business" concern of the plugin.
+
+##### Metrics
+- Throttling applied due lack of data. - Gauge with the value `0` not applied or `1` applied
+- Connection status - Gauge with the value `0` disconnected or `1` Connected
+- publication errors - Counter to allow the detection of connectivity issues.
+- Metrics showing usage per local volume.
+
+##### Configuring the plugin
+###### Existing properties
+Preserved for backwards compatability.
```

Comment:
Could potentially call them deprecated. 

In the single volume case these are easy to support. For the multi volume case they can't be auto migrated sensibly and applying them as global limits while consistent has all of the issues mentioned in the introduction section.